agent/consul/kvs_endpoint.go

```diff
@@ -204,7 +204,10 @@ func (k *KVS) List(args *structs.KeyRequest, reply *structs.IndexedDirEntries) e
 			if err != nil {
 				return err
 			}
+
+			total := len(ent)
 			ent = FilterDirEnt(authz, ent)
+			reply.QueryMeta.ResultsFilteredByACLs = total != len(ent)
```

Comment:
[nit] hm, could be a style preference but curious if we could encapsulate the logic here in a func that can be used for `List` and ListKeys`?

Reply:
Maybe, yeah! Were you thinking of something like `k.filterDirEnt(&reply)` that does the filtering and sets the flag?